packages/apps/reputation-oracle/server/src/modules/kyc/kyc.service.spec.ts

```diff
@@ -1,124 +1,120 @@
+import { faker } from '@faker-js/faker';
 import { createMock } from '@golevelup/ts-jest';
+import { ChainId } from '@human-protocol/sdk';
 import { HttpService } from '@nestjs/axios';
-import { ConfigService } from '@nestjs/config';
 import { Test } from '@nestjs/testing';
+import { ethers } from 'ethers';
 import { of } from 'rxjs';
-import { DeepPartial } from 'typeorm';
-import { MOCK_ADDRESS, mockConfig } from '../../../test/constants';
-import { HCaptchaConfigService } from '../../config/hcaptcha-config.service';
+
 import { KycConfigService } from '../../config/kyc-config.service';
-import { KycStatus } from '../../common/enums/user';
+import {
+  Web3ConfigService,
+  Web3Network,
+} from '../../config/web3-config.service';
+import { KycStatus } from '../kyc/constants';
+import { generateEthWallet } from '../../../test/fixtures/web3';
+import { UserEntity } from '../user';
 import { Web3Service } from '../web3/web3.service';
+import { UpdateKycStatusDto } from './kyc.dto';
 import { KycEntity } from './kyc.entity';
+import { KycError, KycErrorMessage } from './kyc.error';
 import { KycRepository } from './kyc.repository';
 import { KycService } from './kyc.service';
-import { KycError, KycErrorMessage } from './kyc.error';
-import { Web3ConfigService } from '../../config/web3-config.service';
+
+const mockHttpService = {
+  post: jest.fn(),
+};
+
+const mockKycConfigService = {
+  apiKey: faker.string.alphanumeric(),
+  baseUrl: faker.internet.url(),
+};
+
+const operatorWallet = generateEthWallet();
 
 describe('Kyc Service', () => {
   let kycService: KycService;
   let httpService: HttpService;
   let kycRepository: KycRepository;
```

Comment:
You can use `let kycRepository: jest.Mocked<KycRepository>;` type and simplify your spies below because you already rely on `createMock` utility that provides mocked version of class